query/src/plan/stringset.rs

```diff
@@ -0,0 +1,257 @@
+use std::sync::Arc;
+
+use arrow_deps::{datafusion::logical_plan::LogicalPlan, util::str_iter_to_batch};
+use data_types::TABLE_NAMES_COLUMN_NAME;
+
+use crate::{
+    exec::stringset::{StringSet, StringSetRef},
+    util::make_scan_plan,
+};
+
+use snafu::{ResultExt, Snafu};
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Internal error converting to arrow: {}", source))]
+    InternalConvertingToArrow {
+        source: arrow_deps::arrow::error::ArrowError,
+    },
+
+    #[snafu(display("Internal creating a plan for stringset: {}", source))]
```

Comment:
To be consistent with the other error's display:

```suggestion
    #[snafu(display("Internal error creating a plan for stringset: {}", source))]
```